errors/errctx_test.go

```diff
@@ -0,0 +1,153 @@
+package errors
+
+import (
+	"context"
+	stdErrors "errors"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/errgo.v1"
+)
+
+func TestErrCtx_RootCtx(t *testing.T) {
+	t.Run("It should return a default context if the error is not wrapped with ErrCtx", func(t *testing.T) {
```

Comment:
IMO we should return nil in case of no `ErrCtx` wrapped in the given err

Reply:
I'm adding @johnsudaar to the loop since I need an external point of view on this API, I don't feel super super confortable with the error in `RootCtx`

Reply:
FYI, with the current implementation, it could be used like this: https://github.com/Scalingo/go-handlers/pull/57/files#diff-683abe1277a9083805e9aeb9965a862d1d5e422c500ed7d9a41f61e75f306f9aR42-R44
But I share your feel, I'm not comfortable with the error too.

Reply:
I'm not found of this error either.

Maybe another implementation could be changing the method signature ?
```golang
func RootCtx(ctx context.Context, err error) (context.Context)
```

And send the context passed as argument if the context is not found in the error.

```golang
if err != nil {
  logger.Get(errctx.RootCtx(ctx, err)).WithError(err).Error("Something bad happened)
}
```

Reply:
Ok, I like @johnsudaar 's idea, thanks a lot for the help

Maybe in this case `RootCtx` is not the right name, or `err` should be first, since `ctx` is the "fallback" context.

```
errctx.RootCtxOrFallback(err, ctx)
```

(Simple proposal

Reply:
Yep but it's annoying to have linter errors to ignore haha, it won't be the last because of this